flutter/lib/src/navigation/sentry_navigator_observer.dart

```diff
@@ -100,6 +115,32 @@ class SentryNavigatorObserver extends RouteObserver<PageRoute<dynamic>> {
       });
     }
   }
+
+  void _startTransaction(String? name, Object? arguments) {
+    if (!_enableAutoTransactions) {
+      return;
+    }
+    if (name == null) {
+      return;
+    }
+    _transaction = _hub.startTransaction(
+      name,
+      'ui.load',
+      bindToScope: true,
+    );
+    if (arguments != null) {
+      _transaction?.setData('route_settings_arguments', arguments);
+    }
+    _idleTimer = Timer(Duration(seconds: 3), () async {
+      await _finishTransaction();
+    });
```

Comment:
h: All route transactions will have the duration of 3 seconds? Or the transaction is being finished in another place?

Reply:
> All route transactions will have the duration of 3 seconds?

Yes, at least that's how I read it, too. There's no concept of `wait for child spans`  or `idle transactions` like in JS or Android in the Dart SDK yet, so the current code is the only way to create such a transaction for now.

Reply:
I believe this is a hardcoded/raw version of the idle transactions, that should be properly done yet

Reply:
It's a starting point for now, the goal would be for child span to "trim" or extend (`wait for child spans`) the time of the if the last one is finished and no new ones are added.

Reply:
Probably better to also include it in this PR.

Reply:
Can we use 'WidgetsBinding.instance.addPostFrameCallback' inside the observer to now how much time the next 'build' call took?

Reply:
We did discuss using the `addPostFrameCallback`, but came to the conclusion to not use it. @ueman hat some great insight on why not to use it which can be found  [here](https://github.com/getsentry/sentry-mobile/pull/220#issuecomment-965509724) and which ultimately lead us to the current approach.

Reply:
There's also a lot of discussion on https://github.com/getsentry/sentry-dart/pull/611

Reply:
Thanks!